framework/src/NormalisedSumPDF.cpp

```diff
@@ -293,121 +304,81 @@ bool NormalisedSumPDF::SetPhysicsParameters( ParameterSet * NewParameterSet )
 //Return the integral of the function over the given boundary
 double NormalisedSumPDF::Normalisation( DataPoint* NewDataPoint, PhaseSpaceBoundary * NewBoundary )
 {
-	(void)NewDataPoint;
-	(void)NewBoundary;
-	//The evaluate method already returns a normalised value
+	// Calculate and cache the integrals
+	double firstIntegralResult = firstPDF->Integral( NewDataPoint, NewBoundary ) * firstIntegralCorrection;
+	double secondIntegralResult = secondPDF->Integral( NewDataPoint, NewBoundary ) * secondIntegralCorrection;
+	SetCachedIntegrals({firstIntegralResult,secondIntegralResult}, NewDataPoint, NewBoundary);
```

Comment:
No, I think a full equivalence check is needed. The ID will not suffice here as the object can legitimately be different.